openvdb/openvdb/tree/RootNode.h

```diff
@@ -966,6 +978,10 @@ class RootNode
 
     MapType mTable;
     ValueType mBackground;
+#if OPENVDB_ABI_VERSION_NUMBER >= 9
+    /// Transient Data (not serialized)
+    Index32 mTransientData = 0;
+#endif
```

Comment:
@danrbailey as discussed, could you clarify why this is needed on the RootNode

Reply:
The RootNode _is_ a node. This change adds a 32-bit transient integer to every node for use with tools like `NodeManager` and `tools::visitNodesDepthFirst` that visit every node. Why would we exclude the RootNode?

Reply:
> Why would we exclude the RootNode?

Because the point of this change was to take advantage of the space we know is in use? I don't know what 32bits of storage on the RootNode really gives us...

Reply:
It gives us consistency across all node types in the tree hierarchy. One benefit is generic programming - you can write an algorithm that calls `NodeT::transientData()` without having to have one method for InternalNode and LeafNode and another for the RootNode which doesn't have this method. 

This change increases the memory footprint of a tree by just 4 bytes so I really don't see the problem.

Reply:
I'm not sure I agree with this - you already have to specialise RootNode iteration on iter types and things like node masks so the interface is not identical to Internal/Leaf Nodes. The size is irrelevant to me, could be 1 bit! Still seems redundant (to be honest I also don't see the need for it on the internal nodes, but if it's utilising used space then it doesn't matter). Perhaps @jmlait @kmuseth could comment.

Reply:
Lots of things are different between the RootNode, InternalNode and LeafNode, but I frequently don't need to specialize for the RootNode because I'm only using the things that are the same. I ran into a case recently with some changes to the VDB Visualize SOP where I had to build and maintain an external array that would have been simpler and faster if I had a 32-bit integer per-node.

In my mind this is intended to be a flexible mechanism that we can't know exactly how people might wish to use it. Seeing as the cost (4 bytes) is so low, I see no reason not to offer full flexibility rather than just constrain this to leaf nodes. 

I don't think we have a zero redundancy policy for the tree hierarchy - after all the origin coordinates are technically redundant.

Reply:
As agreed in our last meeting, we'll add the 32-bit index to the RootNode as well.